hip-service/Link/Patient/PatientVerification.cs

```diff
@@ -1,27 +1,84 @@
-#nullable enable
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
 using System.Threading.Tasks;
 using hip_service.OTP;
 using HipLibrary.Patient.Model.Response;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Optional;
 
 namespace hip_service.Link.Patient
 {
     public class PatientVerification: IPatientVerification
     {
-        private readonly OtpVerification otpVerification;
-
-        public PatientVerification(OtpVerification otpVerification)
+        private readonly IConfiguration configuration;
+        
+        public PatientVerification(IConfiguration configuration)
         {
-            this.otpVerification = otpVerification;
+            this.configuration = configuration;
         }
-
-        public async Task<Error?> SendTokenFor(Session session)
+        
+        public async Task<OtpMessage> SendTokenFor(Session session)
         {
-            return await otpVerification.GenerateOtp(session);
+            var uri = new Uri(configuration.GetConnectionString("OTPGenerationConnection"));
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(session)).ConfigureAwait(false);
+            return verifyMessage.ValueOr((OtpMessage) null);
         }
         
-        public async Task<Error?> Verify(string sessionId, string value)
+        public async Task<OtpMessage> Verify(string sessionId, string value)
         {
-            return await otpVerification.CheckOtpValue(sessionId, value);
+            var uri = new Uri(configuration.GetConnectionString("OTPVerificationConnection"));
+            var verifyOtpRequest = new OtpVerificationRequest(sessionId, value);
+            var verifyMessage = await PostCallToOTPServer(uri
+                , CreateHttpContent(verifyOtpRequest)).ConfigureAwait(false);
+            return verifyMessage.ValueOr((OtpMessage) null);
```

Comment:
What if there's a failure when it's trying to connect to OTP service?

Reply:
exception is handled in the Post method defined in the same class

Reply:
I do believe returning OtpMessage is not a right option.

Reply:
It is done in the scenario where like our credit runs out and let say TRAI has a rule where you cannot send a message after 9 PM we will come to know about it.